### PR TITLE
#795 Remove unused deps

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #795 

- Not quite as straight forward as I would have liked!
- Removed immer and exports / react table and types
- Removed other incorrect deps. Specifically the ones in `host` caused an issue. Seems host wasn't getting type checked properly, so there was a few type errors that came up

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v16.0.0
```
Comment (joshxg) [inline .nvmrc:1]: i've been using 16, what about you?

File: packages/host/src/Admin/Settings.tsx [hunk 2]
```diff
@@ -37,13 +37,13 @@ export const Settings: React.FC = () => {
       setCustomTheme(themeOptions);
       navigate(0);
     } catch (e) {
-      error(`${t('error.something-wrong')} ${e.message}`)();
```
Comment (joshxg) [inline packages/host/src/Admin/Settings.tsx:40]: some type errors that showed up.. think these are ok though I didn't check too deeply..

File: packages/host/src/components/Footer/Footer.test.tsx [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
-    expect(getByText(/test-user/)).toBeInTheDocument();
-  });
-
-  it('Displays the current store name', () => {
-    const { getByText } = render(
-      <FooterExample username="" storename="test-store" />
-    );
-
-    expect(getByText(/test-store/)).toBeInTheDocument();
+describe('', () => {
```
Comment (joshxg) [inline packages/host/src/components/Footer/Footer.test.tsx:1]: Will sort this out in a PR fixing all the other tests also

**Review by joshxg — COMMENTED** [review]
Also removed the template package. Didn't seem needed

**Comment by mark-prins** [comment]
maybe fixes #935 ?

File: .nvmrc (hunk 1 above)
Comment (mark-prins) [reply .nvmrc:1]: ah. yes. me too

**Comment by mark-prins** [comment]
> Also removed the template package. Didn't seem needed

I was thinking about doing the same.

File: packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonZeroNumberInputCell.tsx [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
 import { CellProps } from '../../../columns';
 import { BasicTextInput } from '@common/components';
-import { DomainObject } from '@common/types';
+import { RecordWithId } from '@common/types';
 import { useBufferState, useDebounceCallback } from '@common/hooks';
 import { NumUtils } from 'packages/common/src/utils/NumUtils';
 
-export const NonZeroNumberInputCell = <T extends DomainObject>({
+export const NonZeroNumberInputCell = <T extends RecordWithId>({
```
Comment (mark-prins) [inline packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonZeroNumberInputCell.tsx:8]: ah, thanks. had also done in the stocktake PR - let's see who gets merged first 😁

File: packages/dashboard/package.json [hunk 5]
```diff
@@ -16,8 +16,6 @@
   "dependencies": {
     "@openmsupply-client/common": "^0.0.1",
     "@openmsupply-client/config": "^0.0.0",
-    "@openmsupply-client/invoices": "^0.0.0",
```
Comment (mark-prins) [inline packages/dashboard/package.json:19]: The `DashboardService` is still using `InboundShipmentWidget` & `OutboundShipmentWidget` from invoices. Shouldn't the dep be in here?

**Review by mark-prins — APPROVED** [review]
👍  Nice work - thanks for that

File: packages/dashboard/package.json (hunk 5 above)
Comment (joshxg) [reply packages/dashboard/package.json]: Thanks 😁

File: packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonZeroNumberInputCell.tsx (hunk 4 above)
Comment (joshxg) [reply packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonZeroNumberInputCell.tsx]: You win 😁

**Comment by joshxg** [comment]
> maybe fixes https://github.com/openmsupply/openmsupply-client/issues/935 ?

woah, whats going on

**Pull request merged**